WalletWasabi/Services/WalletService.cs

```diff
@@ -899,6 +899,21 @@ public async Task<int> CountBlocksAsync()
 				allowedSmartCoinInputs = allowUnconfirmed
 					? Coins.Where(x => !x.Unavailable && allowedInputs.Any(y => y.TransactionId == x.TransactionId && y.Index == x.Index)).ToList()
 					: Coins.Where(x => !x.Unavailable && x.Confirmed && allowedInputs.Any(y => y.TransactionId == x.TransactionId && y.Index == x.Index)).ToList();
+
+				// Add those that have the same script, because common ownership is already exposed.
+				var allScripts = allowedSmartCoinInputs.Select(x => x.ScriptPubKey).ToHashSet();
```

Comment:
You should do this at the CoinSelector level.

Reply:
I know, I just prepared it here.